### ISSUE TITLE
[Feature Request] Upload de Demos GOTV

### ISSUE THREAD

**Issue by paulocrash** [splewis/csgo-pug-setup#239]
I think it would be perfect if you put the option to upload the demos to the internet, just like WarMod:

Configuration of Demos Upload in WarMod:
```
// FTP Upload stuff
wm_autodemoupload_enable "1", "Automatically upload demos when finished recording."
wm_autodemoupload_bzip2 "9", "Compression level. If set > 0 demos will be compressed before uploading. (Requires bzip2 extension.)"
wm_autodemoupload_delete "0", "Delete the demo (and the bz2) if upload was successful."
wm_autodemoupload_ftptargetdemo "demos", "The ftp target to use for uploads."
wm_autodemoupload_ftptargetlog "logs", "The ftp target to use for uploads."
wm_autodemoupload_completed "1", "Only upload demos when match is completed."
```

**Comment by usernameproof** [comment]
While this isn't implemented, I made a quick bash script to do this if anyone's interested.

inotify.sh
```
#!/bin/sh
while true
do
      inotifywait -e close_write --exclude=".txt" /path/to/csgo && sleep 1m && /path/to/sftp.sh
done
```

sftp.sh
```
#!/bin/bash
sftp username@host << DELIMITER
cd path/to/uploaddir
put /path/to/csgo/*.dem
quit
DELIMITER
rm -rf /path/to/csgo/*.dem
```

Make the script run on boot by adding to rc.local
`nohup ./path/to/inotify.sh >/dev/null 2>&1 &`

Requirements: inotify-tools and SSH public-key authentication setup.

**Comment by MoritzLoewenstein** [comment]
since most csgo server providers dont allow full access to bash scripts, you can use [this](https://github.com/MoritzLoewenstein/AutoDemoUpload). This is basically tEasyFtp and tAutoDemoUpload by "Thrawn" with some slight adjustments/fixes (original version dead).

**Comment by crashzk** [comment]
> since most csgo server providers dont allow full access to bash scripts, you can use [this](https://github.com/MoritzLoewenstein/AutoDemoUpload). This is basically tEasyFtp and tAutoDemoUpload by "Thrawn" with some slight adjustments/fixes (original version dead).

That would be your version then, correct?

https://github.com/thraaawn/tEasyFTP

Yours in the case automatically sends the demos to FastDL, is that it?

**Comment by MoritzLoewenstein** [comment]
Yes, this and [this](https://forums.alliedmods.net/showthread.php?p=1517461), as you can see both are not longer supported and didnt compile when I downloaded them. My version compiles fine and works the same, you have to enter fastdl data in addons/sourcemod/configs/RemoteTargets.cfg. I will add a ReadMe.

**Comment by WardPearce** [comment]
Might be worth checking out https://github.com/WardPearce/SQLMatches
